dd-sdk-android/src/test/kotlin/com/datadog/android/rum/gestures/DatadogGesturesListenerTest.kt

```diff
@@ -0,0 +1,492 @@
+            mockRumTracer,
+            WeakReference(decorView)
+        )
+
+        // when
+        underTest.onSingleTapUp(mockEvent)
+
+        // then
+        verify(mockRumTracer).buildSpan(DatadogGesturesListener.UI_TAP_ACTION_EVENT)
+        verify(mockSpanBuilder).withTag(
+            DatadogGesturesListener.TAG_TARGET_CLASS_NAME,
+            decorView.javaClass.canonicalName
+        )
+        verify(mockSpanBuilder).withTag(
+            DatadogGesturesListener.TAG_TARGET_RESOURCE_ID,
+            decorView.id
+        )
+        verify(mockSpan).finish(DatadogGesturesListener.DEFAULT_EVENT_DURATION)
+    }
+
+    @Test
+    fun `onTap adds the target id if any exception while requesting resource id`(forge: Forge) {
+        // given
+        val mockEvent = mockMotionEvent(forge)
+        val targetId = forge.anInt()
+        val validTarget: View = mock(
+            id = targetId,
+            forEvent = mockEvent,
+            hitTest = true
+        )
+        decorView = mock<ViewGroup>(
+            id = forge.anInt(),
+            forEvent = mockEvent,
+            hitTest = false
+        ) {
+            whenever(it.childCount).thenReturn(1)
+            whenever(it.getChildAt(0)).thenReturn(validTarget)
+        }
+        val mockResources = mock<Resources> {
+            whenever(it.getResourceEntryName(validTarget.id))
+                .thenThrow(Resources.NotFoundException(forge.anAlphabeticalString()))
+        }
+        whenever(validTarget.resources).thenReturn(mockResources)
+        underTest = DatadogGesturesListener(
+            mockRumTracer,
+            WeakReference(decorView)
+        )
+
+        // when
+        underTest.onSingleTapUp(mockEvent)
+
+        // then
+        verify(mockRumTracer).buildSpan(DatadogGesturesListener.UI_TAP_ACTION_EVENT)
+        verify(mockSpanBuilder).withTag(
+            DatadogGesturesListener.TAG_TARGET_CLASS_NAME,
+            validTarget.javaClass.canonicalName
+        )
+        verify(mockSpanBuilder).withTag(
+            DatadogGesturesListener.TAG_TARGET_RESOURCE_ID,
+            targetId
+        )
+        verify(mockSpan).finish(DatadogGesturesListener.DEFAULT_EVENT_DURATION)
+    }
+
+    @Test
+    fun `will not send any span if decor view does not have a strong reference`(forge: Forge) {
+        // given
+        val mockEvent = mockMotionEvent(forge)
+        underTest = DatadogGesturesListener(mockRumTracer, WeakReference<View>(null))
+
+        // when
+        underTest.onSingleTapUp(mockEvent)
+
+        // then
+        verifyZeroInteractions(mockRumTracer)
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun mockMotionEvent(forge: Forge): MotionEvent {
+        return mock {
+            whenever(it.x).thenReturn(forge.aFloat(min = 0f, max = XY_MAX_VALUE))
+            whenever(it.y).thenReturn(forge.aFloat(min = 0f, max = XY_MAX_VALUE))
+        }
+    }
+
+    private inline fun <reified T : View> mock(
```

Comment:
you should rename this as `mockView` to avoid confusion with the standard `mock` method.